newa/cli.py

```diff
@@ -194,62 +194,165 @@ def cmd_event(ctx: CLIContext, errata_ids: list[str]) -> None:
-    jira_id_gen = itertools.count(start=1)
-
     for erratum_job in ctx.load_erratum_jobs('event-'):
+
         # read Jira issue configuration
         config = ErratumConfig.from_yaml_file(Path(os.path.expandvars(issue_config)))
 
-        # TODO: record created/existing issues. Instead of `Any`, maybe something
-        # from the Jira library would be stored. Or just a Jira ticket ID.
-        known_issues: dict[str, Any] = {}
+        jira = IssueHandler(config.project, config.transitions)
+        ctx.logger.info(f"Initialized {jira}")
 
-        # Iterate over issue actions. Take one, if it's not possible to finish it,
-        # put it back at the end of the queue.
+        # All issue action from the configuration.
         issue_actions = config.issues[:]
 
+        # Processed action (action.id : issue).
+        processed_actions: dict[str, Issue] = {}
+
+        # Length of the queue the last time issue action was processed,
+        # Use to prevent endless loop over the issue actions.
+        endless_loop_check: dict[str, int] = {}
+
+        # Iterate over issue actions. Take one, if it's not possible to finish it,
+        # put it back at the end of the queue.
         while issue_actions:
             action = issue_actions.pop(0)
 
-            print(f'* Would create a {action.type.name} issue:')
-            print(f'     summary: {action.summary}')
-            print(f'     summary: {action.description}')
-            print()
+            rendered_summary = render_template(action.summary, ERRATUM=erratum_job.erratum)
+            rendered_description = render_template(action.description, ERRATUM=erratum_job.erratum)
+            rendered_assignee = render_template(action.assignee, ERRATUM=erratum_job.erratum)
+
+            ctx.logger.info(f"Processing {action.id} for the current respin")
+            ctx.logger.info(f"  summary     = {rendered_summary}")
+            ctx.logger.info(f"  description = {rendered_description}")
+            ctx.logger.info(f"  assignee    = {rendered_assignee}")
 
-            if action.id in known_issues:
-                raise Exception(f'Issue "{action.id}" is already created!')
+            # Detect that action has parent available (if applicable), if we went trough the
+            # actions already and parent was not found, we abort.
+            if action.parent_id and action.parent_id not in processed_actions:
+                queue_length = len(issue_actions)
+                last_queue_length = endless_loop_check.get(action.id, 0)
+                if last_queue_length == queue_length:
+                    raise Exception(f"Parent {action.parent_id} for {action.id} not found!")
 
-            if action.parent_id and action.parent_id not in known_issues:
-                print(f'     !! Parent issue, "{action.parent_id}", is unknown, will try later')
-                print()
+                endless_loop_check[action.id] = queue_length
```

Comment:
I believe we do not have `action.id` set when it is not present in a config file and it is not a mandatory key (I think it should not be). Therefore, we should probably generate some uniqu `id` in the class constructor if it is not provided as an argument.

Reply:
Alright, I'll come up with some default in IssueAction class (although for better error handling it is rather useful to have it).

Reply:
ack